### PR TITLE
add initial tests with chefspec, foodcritic and rubocop

### PR DESCRIPTION
**Pull request by stevendgonzales**

### CODE REVIEW

File: spec/unit/recipes/default_spec.rb [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+      baseurl: chef_run.node['barbican']['yum_repo']['baseurl'],
+      enabled: true,
+      gpgcheck: chef_run.node['barbican']['yum_repo']['gpgcheck'],
+      gpgkey: chef_run.node['barbican']['yum_repo']['gpgkey'],
+      action: [:create]
+    )
+    
+  end 
+
+  # test that by defualt postgres drivers are not installed
```
Comment (reaperhulk) [inline spec/unit/recipes/default_spec.rb:20]: typo here (super important right?)

File: spec/unit/recipes/default_spec.rb [hunk 2]
```diff
@@ -0,0 +1,54 @@
+require 'spec_helper'
+
+describe 'barbican::default' do
+  
+  let(:chef_run) { ChefSpec::Runner.new.converge(described_recipe) }
+  
+  # barbican repo created
+  it "creates yum repository barbican" do
```
Comment (reaperhulk) [inline spec/unit/recipes/default_spec.rb:8]: Another incredibly minor style nit: this is double quoted while the rest of the rspec test naming in this file is done with single quotes.

**Comment by reaperhulk** [comment]
Other than my super minor comments this is :+1: from me.

**Comment by jfwood** [comment]
:+1:  Nice start to the testing process.

**Pull request merged**